src/java_bytecode/java_types.cpp

```diff
@@ -167,6 +167,243 @@ exprt java_bytecode_promotion(const exprt &expr)
     return typecast_exprt(expr, new_type);
 }
 
+std::vector<typet> parse_list_types(
```

Comment:
shouldn't this be declared in the header file?

Reply:
This is why I prefer classes even without state, as it allows declaring in the header without making a global function of everything that includes `java_types.h` I will move the declarations to the top of the file for now and raise a ticket to refactor this parsing into a class if this is satisfactory for you?

Reply:
Done and raised as [[TG-1206](https://diffblue.atlassian.net/browse/TG-1206)]